app/src/main/java/org/sopt/carrot/data/repositoryimpl/SearchRepositoryImpl.kt

```diff
@@ -0,0 +1,18 @@
+package org.sopt.carrot.data.repositoryimpl
+
+import org.sopt.carrot.data.model.SearchProductModel
+import org.sopt.carrot.data.service.SearchService
+import org.sopt.carrot.domain.repository.SearchRepository
+import org.sopt.carrot.data.mapper.SearchMapper.toSearchProductModel
+
+class SearchRepositoryImpl(
+    private val searchService: SearchService
+): SearchRepository {
+    override suspend fun getSearchResults(keyword: String?): Result<Pair<List<SearchProductModel>, List<SearchProductModel>>> = runCatching {
```

Comment:
p1: 아까 위의 ResponseSearchDto 분리 작업을 반영한다면 반환타입을 아래와 같이 좀 더 줄일 수 있을거 같습니다!!
물론 이 경우 추가되는 매핑로직을 블럭에 추가 작성해주셔야 합니다!!( `SearchProduct`은 제가 임으로 작성한 네이밍입니다!)
```suggestion
    override suspend fun getSearchResults(keyword: String?): Result<SearchProduct> = runCatching {
            val response = searchService.getSearchProducts(keyword)
        }
```

Reply:
여기도 추가적으로 말해보면 지금 위의 코드리뷰를 반영해서 

```
@Serializable
data class ResponseSearchDto(
    @SerialName("products")
    val products: List<Product>,
    @SerialName("similar_products")
    val similarProducts: List<SimilarProduct>
)
@Serializable
data class Product(
    @SerialName("id")
    val id: Long,
    @SerialName("user_id")
    val userId: Long,
    @SerialName("product_image")
    val productImage: String,
    @SerialName("title")
    val title: String?,
    @SerialName("address")
    val address: String?,
    @SerialName("price")
    val price: String
)
@Serializable
data class SimilarProduct(
    @SerialName("id")
    val id: Long,
    @SerialName("user_id")
    val userId: Long,
    @SerialName("product_image")
    val productImage: String,
    @SerialName("title")
    val title: String?,
    @SerialName("address")
    val address: String?,
    @SerialName("price")
    val price: String
)
```

지금 이렇게 Product, SimiliarProduct로 두개로 나뉘었잖아요? 그러면 이를 담을 model도 두개 필요하겠죵? 그러면 또 model에 연결해줄 mapper도 두개 필요하게 되겠죠!